### PR TITLE
feat: udf `replace` function

### LINKED ISSUE #1920: udf function `replace`
**Describe the feature you'd like**

same as https://dev.mysql.com/doc/refman/8.0/en/replace.html

**Additional context**

### PR DESCRIPTION
**Pull request by aceforeverd**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

close #1920 

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: include/base/string_ref.h [hunk 1]
```diff
@@ -27,11 +27,13 @@ namespace base {
 
 struct StringRef {
     StringRef() : size_(0), data_(nullptr) {}
-    explicit StringRef(const char* str)
+    StringRef(std::nullptr_t) : size_(0), data_(nullptr) {} // NOLINT
+
+    StringRef(const char* str) // NOLINT
         : size_(nullptr == str ? 0 : strlen(str)), data_(str) {}
```
Comment (dl239) [inline include/base/string_ref.h]: nullptr will not match this function

Comment (aceforeverd) [reply include/base/string_ref.h]: fixed

**Comment by aceforeverd** [comment]
hybridse-linux build will expected to fail due to storage limit for all static link. 
The `BUILD_SHARED_LIBS` option for Release profile seems have link problems, I will check in another PR
refer #1932

**Review by dl239 — APPROVED** [review]

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Pull request merged**